2015-Fall/lrusnac/CalculatorV2/CalculatorV2/CalculatorModel.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  CalculatorModel.swift
+//  CalculatorV2
+//
+//  Created by Leonid Rusnac on 08/09/15.
+//  Copyright (c) 2015 Leonid Rusnac. All rights reserved.
+//
+
+import Foundation
+
+class CalculatorModel {
+
+    private enum Op: Printable{
+        case Operand(Double)
+        case UnaryOperator(String, Double -> Double)
+        case BinaryOperator(String, (Double, Double) -> Double)
+        case Constant(String, Double)
+        
+        // equivalent of toString in java
+        var description: String {
+            get {
+                switch self {
+                case .Operand(let operand):
+                    return "\(operand)"
+                case .UnaryOperator(let operatorSymbol, _):
+                    return operatorSymbol
+                case .BinaryOperator(let operatorSymbol, _):
+                    return operatorSymbol
+                case .Constant(let operatorSymbol, _):
+                    return operatorSymbol
+                }
+            }
+        }
+    }
+    
+    private var ops = [Op]()
+    private var knownOps = [String:Op]()
+    
+    
+    init() {
+        func learnOp(op: Op) {
+            knownOps[op.description] = op
+        }
+        
+        learnOp(Op.BinaryOperator("+", +))
+        learnOp(Op.BinaryOperator("×", *))
+        learnOp(Op.BinaryOperator("÷", {$1/$0}))
+        learnOp(Op.BinaryOperator("-", {$1-$0}))
+        
+        learnOp(Op.UnaryOperator("√", sqrt))
+        learnOp(Op.UnaryOperator("sin", sin))
+        learnOp(Op.UnaryOperator("cos", cos))
+        
+        learnOp(Op.Constant("π", M_PI))
+    }
+    
+    func pushNumber(number: Double) {
+        ops.append(Op.Operand(number))
+    }
+    
+    func pushOpNotNumber(keyString: String) {
+        if let con = knownOps[keyString] {
+            ops.append(con)
+        }
+    }
+    
+    func clear() {
+        ops.removeAll(keepCapacity: false)
```

Comment:
And thinking about cleaning up is good thing! But this is not C, and often it is actually faster to not think about pointers, and let the runtime sort it out, that to sort it out by hand.

But we're on a tangent here (which I'll gladly follow some other time, but not on github). I mainly just wanted to check if you'd understood what you were poking at since you explicitly wrote it.